### PR TITLE
Include user membership on events

### PR DESCRIPTION
**Pull request by richvdh**
MSC4115 has now completed FCP, so we can enable it by default and switch to the stable identifier.

(Previously: #17104)

### CODE REVIEW

File: scripts-dev/complement.sh [hunk 1]
```diff
@@ -223,7 +223,6 @@ test_packages=(
     ./tests/msc3930
     ./tests/msc3902
     ./tests/msc3967
-    ./tests/msc4115
```
Comment (richvdh) [inline scripts-dev/complement.sh:226]: https://github.com/matrix-org/complement/pull/725 moves this test to the main test suite.

File: synapse/config/experimental.py [hunk 2]
```diff
@@ -436,10 +436,6 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
                 ("experimental", "msc4108_delegation_endpoint"),
             )
 
-        self.msc4115_membership_on_events = experimental.get(
-            "msc4115_membership_on_events", False
-        )
```
Comment (anoadragon453) [inline synapse/config/experimental.py:441]: This instance of setting the experimental config option can also be removed: https://github.com/element-hq/synapse/blob/b8e23537f27a6e5e92200c0827c9db226f1b7216/docker/complement/conf/workers-shared-extra.yaml.j2#L108

File: synapse/visibility.py [hunk 3]
```diff
@@ -101,8 +100,6 @@ async def filter_events_for_client(
         filter_send_to_client: Whether we're checking an event that's going to be
             sent to a client. This might not always be the case since this function can
             also be called to check whether a user can see the state at a given point.
-        msc4115_membership_on_events: Whether to include the requesting user's
-            membership in the "unsigned" data, per MSC4115.
 
     Returns:
         The filtered events. If `msc4115_membership_on_events` is true, the `unsigned`
```
Comment (anoadragon453) [inline synapse/visibility.py:105]: One more instance of `msc4115_membership_on_events` that can be removed.

**Review by anoadragon453 — APPROVED** [review]
Thank you for doing this!

I've included a couple missed spots below, but otherwise LGTM!

**Pull request merged**